ironfish/src/node.ts

```diff
@@ -145,13 +144,22 @@ export class IronfishNode {
       bootstrapNodes: config.getArray('bootstrapNodes'),
       stunServers: config.getArray('p2pStunServers'),
       webSocket: webSocket,
-      node: this,
       chain: chain,
       metrics: this.metrics,
       hostsStore: hostsStore,
       logger: logger,
       telemetry: this.telemetry,
       incomingWebSocketWhitelist: config.getArray('incomingWebSocketWhitelist'),
+      miningManager: this.miningManager,
+      blocksPerMessage: config.get('blocksPerMessage'),
+      memPool,
+      workerPool,
+      // TODO(hugh, rohan): Clean up after the node clients are created for the wallet
+      // `walletAddPendingTransaction` will be removed once we can pull from the mempool
+      walletAddPendingTransaction: (transaction: Transaction) =>
```

Comment:
We should use events instead of passing lambdas into the constructor.

Reply:
I agree, but as per the TODO this is a temporary measure to make the types work / keep this code change small. It's removed in the subsequent PR #4078